[ez] Fsspec Filesystem ls details should be false

ankitageorge opened the pull request:
Summary: The default action for ls for the local filesystem is with details=False, but this isn't the case for all filesystems (eg. huggingface), so setting details=False explicitly so that the return type of ls is a list of strings, and not a list of dictionaries, which is what it would be with details=True.

Test Plan: tested in notebook

Differential Revision: D74080572




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

ankitageorge commented on torch/distributed/checkpoint/_fsspec_filesystem.py:
```diff
@@ -92,7 +92,7 @@ def rm_file(self, path: Union[str, os.PathLike]) -> None:
 
     def ls(self, path: Union[str, os.PathLike]) -> list[str]:
-        return self.fs.ls(path)
+        return self.fs.ls(path, detail=False)
```
ya, it's not a kwarg, it's part of the actual api https://filesystem-spec.readthedocs.io/en/latest/api.html

joecummings replied:
Oh cool! Then wrt the UX, is this something you think users would want exposed so they can toggle on/off?

ankitageorge replied:
I think we should just keep it off so that the return type is always a list of strings and keep it consistent with the filesystem api call https://github.com/pytorch/pytorch/blob/main/torch/distributed/checkpoint/filesystem.py#L542

joecummings replied:
Got it, that makes sense. I could see some confusing in having the PyTorch FileSystem API not match the Fsspec API, but you would know better if people have brought this up at all.

Either way, could be work a comment just saying that we turn off details in order to only ever get back strings.

joecummings reviewed (approved):
LGTM

ankitageorge committed:
[ez] Fsspec Filesystem ls details should be false (#152693)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/152693

The default action for ls for the local filesystem is with details=False, but this isn't the case for all filesystems (eg. huggingface), so setting details=False explicitly so that the return type of ls is a list of strings, and not a list of dictionaries, which is what it would be with details=True.

Test Plan: tested in notebook

Differential Revision: D74080572

ankitageorge commented:
@pytorchmergebot merge -i
